overlord/snapstate/snapstate.go

```diff
@@ -335,7 +335,7 @@ func removeExtraComponentsTasks(st *state.State, snapst *SnapState, targetRevisi
 	return unlinkTasks, discardTasks, nil
 }
 
-func doInstall(st *state.State, snapst *SnapState, snapsup SnapSetup, compsups []ComponentSetup, flags int, fromChange string, inUseCheck func(snap.Type) (boot.InUseFunc, error)) (*state.TaskSet, error) {
+func doInstall(st *state.State, snapst *SnapState, snapsup SnapSetup, compsups []ComponentSetup, flags int, fromChange string, inUseCheck func(snap.Type) (boot.InUseFunc, error), newModel *asserts.Model) (*state.TaskSet, error) {
```

Comment:
this is not how remodel was designed to work, we should pass along the a devicectx that is either the task one (in which case we get the new model) or one created with the new model depending where we are